src/components/NavBar/NavBarModal.jsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import styles from './NavBarModal.module.scss';
+
+const NavBarModal = (props) => {
+    
+    return props.open ? ReactDOM.createPortal(
+        <div id="modal_navbar" className={styles.transparentModal} onClick={props.onClose}>
+        </div>,
+        document.body
+    ): null;
+}
+
+export default NavBarModal;
```

Comment:
There's no point to make this a separated component as the content is trivial, and it's a part of the nav bar's functionality. I'd rather keep it in the same file, and if possible. avoid making it a component at all.

Reply:
Done.